geonode/services/tasks.py

```diff
@@ -64,15 +66,30 @@ def harvest_resource(self, harvest_job_id):
         _cnt = 0
         while _cnt < 5 and not result:
             try:
-                layer = Layer.objects.get(
-                    alternate=f"{workspace.name}:{harvest_job.resource_id}")
+                layer = None
+                if Layer.objects.filter(alternate=f"{harvest_job.resource_id}").count():
```

Comment:
@afabiani can't we always assiign the workspace instead of having to use this if/else statements?